smserver/smserver.go

```diff
@@ -232,26 +232,28 @@ func (server *SMServer) GetAllStatus(ctx context.Context, req *empty.Empty) (sta
 }
 
 // GetBoardConfigStatus gets current board configuration status.
-func (server *SMServer) GetBoardConfigStatus(context.Context, *empty.Empty) (status *pb.BoardConfigStatus, err error) {
+func (server *SMServer) GetBoardConfigStatus(context.Context, *empty.Empty) (*pb.BoardConfigStatus, error) {
 	return &pb.BoardConfigStatus{VendorVersion: server.boardConfigProcessor.GetBoardConfigInfo()}, nil
 }
 
 // CheckBoardConfig checks new board configuration.
-func (server *SMServer) CheckBoardConfig(ctx context.Context,
-	boardConfig *pb.BoardConfig,
-) (status *pb.BoardConfigStatus, err error) {
-	version, err := server.boardConfigProcessor.CheckBoardConfig(boardConfig.GetBoardConfig())
+func (server *SMServer) CheckBoardConfig(ctx context.Context, boardConfig *pb.BoardConfig) (*empty.Empty, error) {
+	if err := server.boardConfigProcessor.CheckBoardConfig(boardConfig.GetBoardConfig()); err != nil {
+		return &emptypb.Empty{}, aoserrors.Wrap(err)
+	}
 
-	return &pb.BoardConfigStatus{VendorVersion: version}, err
+	return &emptypb.Empty{}, nil
 }
 
 // SetBoardConfig sets new board configuration.
 func (server *SMServer) SetBoardConfig(ctx context.Context, boardConfig *pb.BoardConfig) (ret *empty.Empty, err error) {
```

Comment:
in CheckBoardConfig you are not using the name returned value, maybe here also remove it?